File: 1887-reduction-operations-to-make-the-array-elements-equal/1887-reduction-operations-to-make-the-array-elements-equal.cpp
class Solution {
public:
    int reductionOperations(vector<int>& a) {
        
        int Min = *min_element(a.begin() , a.end());
        int Max = *max_element (a.begin() , a.end());
        vector <int> freq (Max + 5);
        for (int i : a){
                
                freq[i] ++;
        }
        int ans = 0 , cnt {};
        for (int i = freq.size() - 1; i >= 1; i --){
            if (freq[i] >= 1 && i != Min){
                
                ans += freq[i] + cnt;
                cnt += freq[i];
            }
        }
        return ans;
    }
};

File: 1887-reduction-operations-to-make-the-array-elements-equal/1887-reduction-operations-to-make-the-array-elements-equal.cs
public class Solution {
    public int ReductionOperations(int[] a) {
        int Min = int.MaxValue;
        int Max = int.MinValue;
        for (int i = 0;i < a.Length; i ++){
            Min = Math.Min(Min , a[i]);
            Max = Math.Max(Max , a[i]);
        }
        int [] freq = new int [Max + 5];
        Array.Fill (freq , 0);
        
        
        foreach (int i in a)
            freq[i] ++;
        
        int count = 0 , ans = 0;
        for (int i = Max; i > Min; i--){
            if (freq[i] >= 1){
                ans += freq[i] + count;
                count += freq[i];
            }
        }
        
        return ans;
    }
}

